File: 1455_CheckIfAWordOccursAsAPrefixOfAnyWordInASentence.py
class Solution:
    def isPrefixOfWord(self, sentence: str, searchWord: str) -> int:
        lengthOfSentence=len(sentence)
        currentIndex=1
        currentString=""
        for i in range(0, lengthOfSentence, 1):
            if currentString==searchWord:
                return currentIndex
            currentChar=sentence[i]
            if currentChar==' ':
                currentString=""
                currentIndex+=1
            else:
                currentString+=currentChar
        if currentString==searchWord:
            return currentIndex
        return -1

File: 1455_CheckIfAWordOccursAsAPrefixOfAnyWordInASentence.cpp
class Solution {
public:
    int isPrefixOfWord(string sentence, string searchWord) {
        int sz=sentence.size(), idx=1;
        string curr="";
        for(int i=0; i<sz; ++i){
            if(curr==searchWord) return idx;
            if(sentence[i]==' '){
                curr="";
                idx++;
            } else{
                curr+=sentence[i];
            }
        }
        if(curr==searchWord) return idx;
        return -1;
    }
};
